src/features/DeFiZap/stateFactory.tsx

```diff
@@ -44,15 +45,15 @@ const ZapInteractionFactory: TUseStateReducerFactory<ZapInteractionState> = ({
         .then((retrievedTxReceipt) => retrievedTxReceipt)
         .catch((hash) => provider.getTransactionByHash(hash))
         .then((retrievedTransactionReceipt) => {
```

Comment:
Unfortunately, they're not the same thing. as for whether we /need/ both, i believe we need to at least make the call to fetch both at some point.

[TransactionResponse](https://docs.ethers.io/ethers.js/html/api-providers.html#transaction-response) is a sort of a breakdown of a transaction. It can be fetched before tx is included in the block.

[TransactionReceipt](https://docs.ethers.io/ethers.js/html/api-providers.html#transaction-receipt) is the "receipt" of the transaction. It can't be fetched before the tx is included in a block. But we need to call this one because TransactionResponse doesn't contain a `status` field.